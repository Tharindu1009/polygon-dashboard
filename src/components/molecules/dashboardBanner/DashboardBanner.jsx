
import './styles/dashboardBanner.scss';
import * as React from 'react';
import dashboardBannerImage from '../../../assets/images/dashboardBanner.png';

function DashboardBanner({ title, description }) {
    return (
        <div className='dashboardBanner' data-testid="dashboardBanner">
            <div className='description-wrapper'>
                <div className='description-title'>{title}</div>
                <div className='description-content'>{description}</div>
            </div>
            <div className='image-wrapper'>
                <img className='image' src={dashboardBannerImage} alt='Dashboard Banner' />
            </div>
        </div>
    );
}

export default DashboardBanner;

