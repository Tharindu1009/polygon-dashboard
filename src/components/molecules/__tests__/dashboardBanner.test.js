import { render, screen, cleanup } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import DashboardBanner from "../dashboardBanner/DashboardBanner";

afterEach(() => {
    cleanup();
})
const title = "Welcome to your dashboard!";
const description = `Try our new Admin Dashboard Template, 
build with live Ant-Design components. Customize it to your needs and release to production!`

test('should render DashboardBanner molecule', () => {
    render(<DashboardBanner title={title} description={description} />);
    const dashboardBanner = screen.getByTestId('dashboardBanner');
    expect(dashboardBanner).toBeInTheDocument();
    expect(dashboardBanner).toHaveTextContent("Welcome to your dashboard!");
})

test('matches DashboardBanner snapshot', () => {
    const tree = TestRenderer.create(<DashboardBanner title={title} description={description} />).toJSON();
    expect(tree).toMatchSnapshot();
})