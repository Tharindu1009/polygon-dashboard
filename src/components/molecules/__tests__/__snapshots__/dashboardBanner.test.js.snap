// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches DashboardBanner snapshot 1`] = `
<div
  className="dashboardBanner"
  data-testid="dashboardBanner"
>
  <div
    className="description-wrapper"
  >
    <div
      className="description-title"
    >
      Welcome to your dashboard!
    </div>
    <div
      className="description-content"
    >
      Try our new Admin Dashboard Template, 
build with live Ant-Design components. Customize it to your needs and release to production!
    </div>
  </div>
  <div
    className="image-wrapper"
  >
    <img
      alt="Dashboard Banner"
      className="image"
      src="dashboardBanner.png"
    />
  </div>
</div>
`;
